### PR TITLE
CutTool: Fix line cutting operations

### PR DESCRIPTION
**Pull request by lpechacek**
Fix for GH-1227 and GH-1276.

### CODE REVIEW

File: src/tools/cut_tool.cpp [hunk 1]
```diff
@@ -354,7 +357,9 @@ void CutTool::finishCuttingLine()
 	auto split_objects = edit_object->removeFromLine(drag_part_index, qreal(drag_start_len), qreal(drag_end_len));
 	replaceObject(edit_object, split_objects);
 	
-	finishEditing();
+	if (editingInProgress())
```
[the remark this answers is not in the document]
Comment (lpechacek) [reply src/tools/cut_tool.cpp]: No, it doesn't. In the "Cut line from start to end, effectively wiping the entire line." scenario the selection becomes empty and something somewhere in the chain finishes editing. If you want, I can provide a more detailed analysis as I traversed the whole chain while working on the fix. I don't have the exact trace at hand but I can reconstruct it.

[the remark this answers is not in the document]
Comment (lpechacek) [reply src/tools/cut_tool.cpp]: The editing flag setting is not that straightforward as I thought. Lemme dive deeper into this part.

[the remark this answers is not in the document]
Comment (lpechacek) [reply src/tools/cut_tool.cpp]: So this is how `finishEditing()` gets triggered in the "wipe line" case. After  finishing the cursor drag movement covering the whole line the following sequence of actions rolls out:
```
CutTool::dragFinish()
-> CutTool::finishCuttingLine() - split_objects() returns empty set
   -> CutTool::replaceObject() - replaces object with nothing and therefore empties selection
      -> Map::emitSelectionChanged() - emits Map::objectSelectionChanged signal
	 -> MapEditorToolBase::objectSelectionChanged()
	    -> CutTool::objectSelectionChangedImpl() - sees that there are no line or area objects left in selection
	       -> MapEditorTool::deactivate()
		  -> MapEditorController::setEditTool()
		     -> MapEditorController::setTool() 
			-> MapEditorToolBase::finishEditing()
```
As part of this sequence `CutTool::replaceObject()` resumes operation on essentially dead `CutTool` object instance as there was `EditTool` activated in the meantime and `CutTool`'s edit operation finalized. Hope this clarifies why the condition has been introduced.

[the remark this answers is not in the document]
Comment (dg0yt) [reply src/tools/cut_tool.cpp]: ACK

**Pull request merged**